named_arrays/scalars/array_functions.py

```diff
@@ -99,7 +99,7 @@ def array_function_default(
         kwargs['keepdims'] = keepdims
     if initial is not None:
         kwargs['initial'] = initial
-    if where is not None:
+    if where is not None and not np._NoValue:
```

Comment:
Looks like we made a naive error here.

```suggestion
    if where is not None and where is not np._NoValue:
```

Reply:
I just fixed this in `main` since I needed it to work for other tests. Delete this change and merge in from main.